demo/runners/faber.py

```diff
@@ -314,6 +340,70 @@ def generate_proof_request_web_request(
                     proof_request_web_request["connection_id"] = self.connection_id
                 return proof_request_web_request
 
+            elif cred_type == CRED_FORMAT_VC_DI:
+                req_attrs = [
+                    {
+                        "name": "name",
+                        "restrictions": [{"schema_name": "degree schema"}],
+                    },
+                    {
+                        "name": "date",
+                        "restrictions": [{"schema_name": "degree schema"}],
+                    },
+                ]
+                if revocation:
+                    req_attrs.append(
+                        {
+                            "name": "degree",
+                            "restrictions": [{"schema_name": "degree schema"}],
+                            "non_revoked": {"to": int(time.time() - 1)},
+                        },
+                    )
+                else:
+                    req_attrs.append(
+                        {
+                            "name": "degree",
+                            "restrictions": [{"schema_name": "degree schema"}],
+                        }
+                    )
+                if SELF_ATTESTED:
+                    # test self-attested claims
+                    req_attrs.append(
+                        {"name": "self_attested_thing"},
+                    )
+                req_preds = [
+                    # test zero-knowledge proofs
+                    {
+                        "name": "birthdate_dateint",
+                        "p_type": "<=",
+                        "p_value": int(birth_date.strftime(birth_date_format)),
+                        "restrictions": [{"schema_name": "degree schema"}],
+                    }
+                ]
+
+                vc_di_proof_request = {
+                    "name": "Proof of Education",
+                    "version": "1.0",
+                    "requested_attributes": {
+                        f"0_{req_attr['name']}_uuid": req_attr for req_attr in req_attrs
+                    },
+                    "requested_predicates": {
+                        f"0_{req_pred['name']}_GE_uuid": req_pred
+                        for req_pred in req_preds
+                    },
+                }
+
+                if revocation:
+                    vc_di_proof_request["non_revoked"] = {"to": int(time.time())}
+
+                proof_request_web_request = {
+                    "presentation_request": {"vc_di": vc_di_proof_request},
+                    "trace": exchange_tracing,
+                }
+                if not connectionless:
+                    proof_request_web_request["connection_id"] = self.connection_id
+                return proof_request_web_request
```

Comment:
this is good, do we need to add any additional tests to demonstrate that the vc is a valid verifiable credential?  or is that out of scope of our changes?